### ISSUE TITLE
Overnight test is failing: freedesktop-sdk being used need to be updated to support buildstream master

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#1334]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/1334)
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 2, 2020, 11:43

Job [#576177727](https://gitlab.com/BuildStream/buildstream/-/jobs/576177727) failed for 32a531747c45fb96df64da6c7e3514cedeaaf53a:

Update: We've now merged https://gitlab.com/BuildStream/buildstream/-/merge_requests/1987, and expect the overnight tests to start passing. If the overnight job(s) pass on the next schedule, this issue ought to be closed.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 2, 2020, 11:44

changed title from **Overnigth test is failing:** to **Overnigth test is failing:{+ freedesktop-sdk being used need to be updated to support buildstream master+}**

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 2, 2020, 14:33

assigned to [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 2, 2020, 14:41

mentioned in merge request !1953

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 2, 2020, 16:15

the expected target `valentindavid/bst2` will need some work (eg. `pip` is now in bst-plugins-experimental). Some plugins will need to be ported to use the new vdir api also

i'm updating on top of this branch on a fork currently

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 3, 2020, 11:06

assigned to [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 4, 2020, 08:26

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) has been working on modifications to valentindavid/bst2 that will fix more of the branch against bst master.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 9, 2020, 23:54

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) any further progress on this?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) on Jun 10, 2020, 09:03

Umm, I have got FD building a lot further it now gets as far as elements that have plugins that need converting to bst-master (all the plugins can be configured but some still have not been converted to 'virtual'/'cas based' diretories)

I am waiting for buildbox to release a new tag so that i can merge what my work needs to the master branch of FD-docker-images.

This dose not block fixing the plugins but i have been waiting for the new tag. This should be done some time this week but [[Gitlab user @juergbi]](https://gitlab.com/juergbi) is just trying to get a few more bits merged but like i say this is not a hard blocker on working on the plugins.

Please see, https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/jobs/580051634 for details.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 11, 2020, 11:09

plugins are tracked at https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/990

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 19, 2020, 06:15

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) I think a new tag has been released; is anything blocking this? Does a WIP MR exist somewhere?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 19, 2020, 08:33

nothing blocks this afaik except having the time to do the mentioned plugin porting and other cleanups. I believe [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) is working on this?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jun 19, 2020, 08:36

I've had a look at the plugins in question though and i think for most of these it should be a drop in replacement of api

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) on Jun 19, 2020, 08:47

I think there have been some issues caused by what bst can accept as unique keys now, as well. But most of the work should be changing the way the plugins access the sandbox directory. Which in theory should be fairly straight forward. finggers crossed

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jun 19, 2020, 14:33

Can confirm, have been working through these. I've done 4/5.

Updating them to the new API has been fairly straightforward.
Figuring out the new API, when I wasn't even familiar with the old API, has been a lot less straightforward. But I got there.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 28, 2020, 22:34

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) nice, can you please create a WIP: Merge Request with what you have at the moment?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @christopherphang]](https://gitlab.com/christopherphang) on Jun 29, 2020, 11:29

marked this issue as related to freedesktop-sdk/freedesktop-sdk#990

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @christopherphang]](https://gitlab.com/christopherphang) on Jun 29, 2020, 11:29

removed the relation with freedesktop-sdk/freedesktop-sdk#990

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @christopherphang]](https://gitlab.com/christopherphang) on Jun 29, 2020, 11:29

marked this issue as related to freedesktop-sdk/freedesktop-sdk#990

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 29, 2020, 21:03

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) I tried to build the current willsalmon/overnight branch (which uses the bst2 branch from freedesktop-sdk, which seems to be the correct one: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/990#note_369075452) 

But it doesnt' seems to work: https://gitlab.com/BuildStream/buildstream/-/jobs/616390670

Am I missing something?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) on Jun 29, 2020, 21:55

So that looks like on of the plugins ill have a look, bst is not good at reporting which one but the big issue is that the branch [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) and myself are working on for FD-SDK is targeting the bst-1.93.4 dev tag and bst-master has already changed the junction API so im not sure were we are going to find the resources to maintain too many branches..

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 30, 2020, 12:20

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) I would suggest to change the CI job to (temporarily) use that tag; make the CI pass, then other a new issue to move from 1.93.4 to master

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) on Jun 30, 2020, 12:23

[[Gitlab user @jjardon]](https://gitlab.com/jjardon) my branch was based on the 1.93.4 tag, i dont know what you have done with your branch.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jun 30, 2020, 12:35

[[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) no changes in my branch apart from allow the schedule job to run

can you please a MR here so we are sure we are talking about the same change?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 6, 2020, 11:08

I believe everyone in this conversation already knows this, but just to keep the issue up to date:

The closest thing we have to a canonical bst 2 branch of freedesktop-sdk is now [freedesktop-sdk-20.08beta.1-buildstream2](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/tree/freedesktop-sdk-20.08beta.1-buildstream2).

However that branch is incomplete, waiting on https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2851 and https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2853.

The source branch for https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2853 is our current best contender for a complete CI pass on bst 1.93.4, but it still fails on the 32 bit jobs.

[[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) seems to be working on changes to buildbox-fuse, which I think are intended to resolve that. https://gitlab.com/freedesktop-sdk/infrastructure/freedesktop-sdk-docker-images/-/merge_requests/120


[[Gitlab user @jjardon]](https://gitlab.com/jjardon) I didn't completely follow your question, but are you still asking for an MR?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Jul 6, 2020, 16:40

I'll add that freedesktop-sdk/infrastructure/freedesktop-sdk-docker-images!120 is just a workaround for https://gitlab.com/BuildGrid/buildbox/buildbox-fuse/-/issues/40

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 6, 2020, 18:26

mentioned in merge request bst-plugins-experimental!125

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 6, 2020, 18:31

I think I've got the problem with [[Gitlab user @jjardon]](https://gitlab.com/jjardon)'s job resolved. There's a problem in get_unique_key for the flatpak_image plugin.

The fix is here: https://gitlab.com/BuildStream/bst-plugins-experimental/-/merge_requests/125

Compare: https://gitlab.com/BuildStream/buildstream/-/jobs/616390670 [[Gitlab user @jjardon]](https://gitlab.com/jjardon)'s original failing job
With: https://gitlab.com/BuildStream/buildstream/-/jobs/625901528 (The same job, except using the updated branch of bst-plugins-experimental)

The only thing I don't understand, is why it wasn't an issue in the freedesktop-sdk jobs, and why it was an issue here.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 7, 2020, 10:39

UPDATE - We now have a completely successful pipeline that runs the overnight job, using bst2. See the overnight job at: https://gitlab.com/BuildStream/buildstream/-/jobs/625935362

The fix for this particular issue seems to be:
1. Use a branch of bst-plugins-experimental that fixes the unique_key issue in flatpak_image. (This can be master, as soon as https://gitlab.com/BuildStream/bst-plugins-experimental/-/merge_requests/125 merges)
2. Use a branch of freedesktop-sdk that includes the sandbox build-arch fix from https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commit/c956395dfe5634529d74cfe797944a827b37eefb. Currently the best available branch is - the source branch from https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2853. When that that MR merges, we'll be able to use [update-plugins-to-bst-1.93.4](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/tree/dwinship/rebase-bst2/update-plugins-to-bst-1.93.4).

That means we've figured out the technical issues, but I'm not clear what we actually want to do with this. Do we need to maintain a branch where the overnight tests pass? When do we want to merge this into master? etc.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @christopherphang]](https://gitlab.com/christopherphang) on Jul 7, 2020, 13:17

[[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) [[Gitlab user @jjardon]](https://gitlab.com/jjardon) might want to comment, but I would ensure the MR in bst-plugins-experimental gets merged, and merge any required changes to the the 'canonical' bst-2 branch in freedesktop-sdk

Once you have both items, I would prepare an MR in buildstream that will change gitlab-ci and point to those new refs for bst-plugins-experimental and freedesktop-sdk. That MR can be merged to master.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Jul 7, 2020, 15:16

> UPDATE - We now have a completely successful pipeline that runs the overnight job, using bst2. See the overnight job at: https://gitlab.com/BuildStream/buildstream/-/jobs/625935362

Well, not yet. You have a branch on top of 1.93.4 that's successful. Rebasing it on top of master will make it fail again because of !1901

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 7, 2020, 16:30

Oh, to be clear, I meant that we had a pipeline that worked with some version of bst2. I didn't mean to imply that we had one working with bst master.

But now that you mention it, I tried rebasing it on master earlier today. And it looks like it did pretty well: https://gitlab.com/BuildStream/buildstream/-/pipelines/163947859

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 7, 2020, 16:32

Ah, I clearly have selective vision. That pipeline didn't pass at all. I was seeing what I wanted to see.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 7, 2020, 19:10

I have an MR now in freedesktop-sdk, to resolve the junction conflict: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2925

And I'm testing it buildstream's CI jobs here: https://gitlab.com/BuildStream/buildstream/-/merge_requests/1987 (2nd / most recent pipeline)

UPDATE: MR 2925 won't be accepted. The change works well with current buildstream master (see the job above), but it's not compatible with bst 1.93.4 (which the freedesktop-sdk branch currently targets).

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) on Jul 8, 2020, 09:40

[[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) looks like you'll need to rebase that on https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commits/valentindavid/bst2

edit: actually, [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon) you seem to be missing a commit from there on https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commits/willsalmon/bst2 too

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 8, 2020, 11:03

[[Gitlab user @traveltissues]](https://gitlab.com/traveltissues) [freedesktop-sdk-20.08beta.1-buildstream2](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/tree/freedesktop-sdk-20.08beta.1-buildstream2) is the new main branch for buildstream2 development on freedesktop-sdk.

It was built on top of [[Gitlab user @willsalmon]](https://gitlab.com/willsalmon)'s branch, which was built on top of valentindavid's branch. But they've both been rebased on master more recently than valentin's has been. (And beta.1-buildstream2 has been re-ordered and had a few commits squashed together as well).

That might be why there's some commits apparently missing.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Jul 9, 2020, 11:45

mentioned in merge request !1987

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 9, 2020, 11:49

changed title from **Overnig{-th-} test is failing: freedesktop-sdk being used need to be updated to support buildstream master** to **Overnig{+ht+} test is failing: freedesktop-sdk being used need to be updated to support buildstream master**

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Jul 15, 2020, 03:17

closed via merge request !1987

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Jul 15, 2020, 03:17

mentioned in commit a13693ea54fdc16db4d51c7162a71f008df7132d

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @DouglasWinship]](https://gitlab.com/DouglasWinship) on Jul 15, 2020, 10:29

changed the description
